### PR TITLE
Enhancement: Add support to list zones

### PR DESCRIPTION
**Pull request by jackdpeterson**
Feature enhancement: list domains to support bulk operations.
new method:
```
$powerdns->listZones: Zone[];
```

Once working with Zone[], knowing what zone is being returned is helpful for consuming code.
```
$zone->getCanonicalName(): string;
```

### CODE REVIEW

File: src/AbstractZone.php [hunk 1]
```diff
@@ -85,4 +85,12 @@ protected function getZonePath(?string $path = null) : string
     {
         return sprintf('zones/%s%s', $this->zone, $path);
     }
+
+    /**
+     * @return string
+     */
```
Comment (trizz) [inline src/AbstractZone.php]: Please use complete docblocks.

```php
   /**
    * Get the canonnical name of the zone.
    *
    * @return string The canonical zone name.
    */
```

File: src/AbstractZone.php [hunk 2]
```diff
@@ -85,4 +85,12 @@ protected function getZonePath(?string $path = null) : string
...
+    public function getCanonicalName(): string
+    {
+        return substr($this->zone, 0, strlen($this->zone) - 1);
```
Comment (trizz) [inline src/AbstractZone.php]: A canonical name _must_ include the `.` at the end. Either change this line or rename the method.

```suggestion
        return $this->zone;
```

File: src/Powerdns.php [hunk 3]
```diff
@@ -174,6 +174,19 @@ public function deleteZone(string $canonicalDomain) : bool
         return true;
     }
 
+    /**
+     * @return Zone[]
+     */
```
Comment (trizz) [inline src/Powerdns.php]: Please use complete docblocks.
```php
    /**
     * Retrieve all zones.
     *
     * @return Zone[] Array containing the zones.
     */
```

File: tests/ZoneTest.php [hunk 4]
```diff
@@ -112,4 +112,11 @@ public function testFindResourceRecords() : void
         $this->assertSame(1, $zone->find('record01.test.nl.')->count());
         $this->assertSame(0, $zone->find('record01.test.nl.', 'MX')->count());
     }
+
+    public function testGetCanonicalName(): void
+    {
+        $connector = Mockery::mock(Connector::class);
+        $zone = new Zone($connector, 'test.nl');
+        $this->assertSame('test.nl', $zone->getCanonicalName());
```
Comment (trizz) [inline tests/ZoneTest.php]: A canonical name should end with a `.` (see my previous comment).

```suggestion
        $this->assertSame('test.nl.', $zone->getCanonicalName());
```

**Comment by jackdpeterson** [comment]
Thanks for the review! I just implemented those docblock changes and modified the behavior of getCanoncalName to return the $this->zone.

**Comment by robbinjanssen** [comment]
Released as [v1.1.0](https://github.com/exonet/powerdns-php/releases/tag/v1.1.0) :tada:

**Pull request merged**